### ISSUE TITLE
android.database.sqlite.SQLiteCantOpenDatabaseException: unable to open database file

### ISSUE THREAD

**Issue by mntgoat** [facebook/stetho#154]
Just got this over crashlytics. I'm guessing it happened to me because I only initialize Stetho if the app is debuggable but I don't remember the crash. Anyways I figured I should report it. I am using Stetho 1.1

Thanks. 

```
android.database.sqlite.SQLiteCantOpenDatabaseException: unable to open database file
       at android.database.sqlite.SQLiteDatabase.dbopen(SQLiteDatabase.java)
       at android.database.sqlite.SQLiteDatabase.openDatabase(SQLiteDatabase.java:1013)
       at android.database.sqlite.SQLiteDatabase.openDatabase(SQLiteDatabase.java:986)
       at android.database.sqlite.SQLiteDatabase.openDatabase(SQLiteDatabase.java:962)
       at com.facebook.stetho.inspector.database.DatabasePeerManager.openDatabase(DatabasePeerManager.java:210)
       at com.facebook.stetho.inspector.database.DatabasePeerManager.getDatabaseTableNames(DatabasePeerManager.java:120)
       at com.facebook.stetho.inspector.protocol.module.Database.getDatabaseTableNames(Database.java:86)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:511)
       at com.facebook.stetho.inspector.MethodDispatcher$MethodDispatchHelper.invoke(MethodDispatcher.java:98)
       at com.facebook.stetho.inspector.MethodDispatcher.dispatch(MethodDispatcher.java:69)
       at com.facebook.stetho.inspector.ChromeDevtoolsServer.handleRemoteRequest(ChromeDevtoolsServer.java:131)
       at com.facebook.stetho.inspector.ChromeDevtoolsServer.handleRemoteMessage(ChromeDevtoolsServer.java:113)
       at com.facebook.stetho.inspector.ChromeDevtoolsServer.onMessage(ChromeDevtoolsServer.java:89)
       at com.facebook.stetho.websocket.WebSocketSession$1.handleTextFrame(WebSocketSession.java:178)
       at com.facebook.stetho.websocket.WebSocketSession$1.onCompleteFrame(WebSocketSession.java:138)
       at com.facebook.stetho.websocket.ReadHandler.readLoop(ReadHandler.java:48)
       at com.facebook.stetho.websocket.WebSocketSession.handle(WebSocketSession.java:47)
       at com.facebook.stetho.websocket.WebSocketHandler.doUpgrade(WebSocketHandler.java:156)
       at com.facebook.stetho.websocket.WebSocketHandler.handleSecured(WebSocketHandler.java:108)
       at com.facebook.stetho.server.SecureHttpRequestHandler.handle(SecureHttpRequestHandler.java:54)
       at org.apache.http.protocol.HttpService.doService(HttpService.java:243)
       at org.apache.http.protocol.HttpService.handleRequest(HttpService.java:187)
       at com.facebook.stetho.server.LocalSocketHttpServer$WorkerThread.run(LocalSocketHttpServer.java:233)

```

**Comment by jasta** [comment]
Hmm.  You must have been inspecting actively in stetho to cause this, correct?  I'm really not sure why this would happen in practice tho.  I'll have to dig in a bit when I get back from Droidcon

**Comment by jasta** [comment]
Looking closer, this can clearly happen due to races (such as removing a database shortly after Stetho begins initialization).  We soften this exception in other areas (such as executeSQL) and I think the right call is to soften it for getDatabaseTableNames as well.

**Comment by mntgoat** [comment]
You know I did have some crashes of the Chrome window while I was doing the inspection but I don't remember the app crashing. Anyways what I did to get those Chrome window crashes is I looked into the preferences of other libraries under the local storage section. I first looked at my own preferences and then I thought, ohh look at all those other preferences, and as soon as I clicked on one of those the window just closed itself.
